client/router_create.go

```diff
@@ -86,17 +86,19 @@ func EdgeListener(options types.SiteConfigSpec) qdr.Listener {
 }
 
 func (cli *VanClient) getControllerRules() []rbacv1.PolicyRule {
-	if cli.RouteClient == nil {
-		// remove rule for routes if routes not defined
-		rules := []rbacv1.PolicyRule{}
-		for _, rule := range types.ControllerPolicyRule {
-			if len(rule.APIGroups) > 0 && rule.APIGroups[0] != "route.openshift.io" {
-				rules = append(rules, rule)
-			}
+	// remove rule for routes or DeploymentConfigs if they are not defined
+	var rules []rbacv1.PolicyRule
+	for _, rule := range types.ControllerPolicyRule {
+		shouldAvoidRoutesRule := cli.RouteClient == nil && len(rule.APIGroups) > 0 && rule.APIGroups[0] == "route.openshift.io"
+		shouldAvoidDepConfigRule := cli.OCAppsClient == nil && len(rule.APIGroups) > 0 && rule.APIGroups[0] == "apps.openshift.io"
+
```

Comment:
Another minor stylistic point, which may just be personal preference, but for me this would be clearer as two separate if statements, since the only code executed if true is `continue`.

Reply:
Updated in 15e7392085250976000128be87742e0763e27921